packages/cli/src/tokens/write.ts

```diff
@@ -26,45 +31,77 @@ const generateTypographyFile = (
 ): File => {
   const path = `${outPath}/primitives/modes/typography/${folder}`;
   return {
-    data: JSON.stringify(tokens, null, 2),
+    data: stringify(tokens),
     path,
     filePath: `${path}/${name}.json`,
   };
 };
 
-export const writeTokens = async (writeDir: string, tokens: Tokens) => {
+type WriteTokensOptions = {
+  writeDir: string;
+  tokens: Tokens;
+  themeName: string;
+};
+
+export const writeTokens = async (options: WriteTokensOptions) => {
+  const { writeDir, tokens, themeName } = options;
   const targetDir = path.resolve(process.cwd(), String(writeDir));
+  const $themesPath = path.join(targetDir, '$themes.json');
+  const $metadataPath = path.join(targetDir, '$metadata.json');
+  let themes = [themeName];
+
   await fs.mkdir(targetDir, { recursive: true });
 
-  // Generate metadata and themes json for Token Studio and build script
-  console.log('Generating metadata and themes files');
-  const $theme = generateThemesJson(['light', 'dark', 'contrast'], ['theme']);
-  const $metadata = generateMetadataJson(['light', 'dark', 'contrast'], ['theme']);
+  try {
+    // Update with existing themes
+    const $themes = await fs.readFile($themesPath, 'utf-8');
+    const themeObjects = (JSON.parse($themes) as ThemeObject[]) || [];
+    themes = R.pipe(
+      R.filter((obj: ThemeObject) => R.toLower(obj.group || '') === 'theme'),
+      R.map(R.prop('name')),
+      R.append(themeName),
+      R.uniq,
+    )(themeObjects) as string[];
+  } catch (error) {}
```

Comment:
We are exclusively looking for `$themes.json` here so figured it would be ok. The alternative is to use [fs.access](https://nodejs.org/api/fs.html#fs_fs_access_path_mode_callback), but I figured this would be ok for now 😅

Reply:
Yeah if you don't have read access to $themes.json you probably don't have write access either (although it's theoretically possible)